dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+                                return true;
+                            }
+                        }
+                    }
+                    focusedIndex += indexIncrement;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+bool Breadcrumb::MoveFocusPrevious()
+{
+    int movementPrevious{ -1 };
+
+    // If the focus is in the first visible item, then move to the ellipsis
+    if (const auto& repeater = m_breadcrumbItemRepeater.get())
+    {
+        const auto& repeaterLayout = repeater.Layout();
+        if (const auto& breadcrumbLayout = repeaterLayout.try_as<BreadcrumbLayout>())
+        {
+            if (breadcrumbLayout->EllipsisIsRendered() &&
+                m_focusedIndex == static_cast<int>(breadcrumbLayout->FirstRenderedItemIndexAfterEllipsis()))
+            {
+                movementPrevious = -m_focusedIndex;
+            }
+        }
+    }
+
+    return MoveFocus(movementPrevious);
+}
+
+bool Breadcrumb::MoveFocusNext()
+{
+    int movementNext{ 1 };
+
+    // If the focus is in the ellipsis, then move to the first visible item 
+    if (m_focusedIndex == 0)
+    {
+        if (const auto& repeater = m_breadcrumbItemRepeater.get())
+        {
+            const auto& repeaterLayout = repeater.Layout();
+            if (const auto& breadcrumbLayout = repeaterLayout.try_as<BreadcrumbLayout>())
+            {
+                movementNext = breadcrumbLayout->FirstRenderedItemIndexAfterEllipsis();
+            }
+        }
+    }
+
+    return MoveFocus(movementNext);
+}
+
+// If we haven't handled the key yet and the original source was the first(for up and left)
+// or last(for down and right) element in the repeater we need to handle the key so
+// RadioButton doesn't, which would result in the behavior.
+bool Breadcrumb::HandleEdgeCaseFocus(bool first, const winrt::IInspectable& source)
+{
+    if (auto const& repeater = m_breadcrumbItemRepeater.get())
+    {
+        if (auto const& sourceAsUIElement = source.try_as<winrt::UIElement>())
+        {
+            auto const index = [first, repeater]()
+            {
+                if (first)
+                {
+                    return 0;
+                }
+                if (auto const& itemsSourceView = repeater.ItemsSourceView())
+                {
+                    return itemsSourceView.Count() - 1;
+                }
+                return -1;
+            }();
+
+            if (repeater.GetElementIndex(sourceAsUIElement) == index)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+winrt::FindNextElementOptions Breadcrumb::GetFindNextElementOptions()
+{
+    auto const& findNextElementOptions = winrt::FindNextElementOptions{};
+    findNextElementOptions.SearchRoot(*this);
+    return findNextElementOptions;
+}
+
+void Breadcrumb::OnChildPreviewKeyDown(const winrt::IInspectable&, const winrt::KeyRoutedEventArgs& args)
+{
+    switch (args.Key())
```

Comment:
Somebody may use 
`    <local:BreadcrumbLayout Orientation='Vertical'/>`
in their Breadcrumb control template.

In that case the Up/Down keys should be processed like Left/Right, etc...

Reply:
Given Ranjesh's comment about using NonVirtualizingLayout, the vertical orientation becomes a non-issue.